src/java/picard/illumina/IlluminaBasecallsToSam.java

```diff
@@ -206,6 +206,16 @@
             "BAMs for only a subset of the barcodes in a lane.", shortName="INGORE_UNEXPECTED")
     public boolean IGNORE_UNEXPECTED_BARCODES = false;
 
+    @Option(doc="The tag to use to store any molecular indexes.  If more than one molecular index is found, they will be concatenated and stored here.")
+    public String MOLECULAR_INDEX_TAG = "RX";
+
+    @Option(doc="The tag to use to store any molecular index base qualities.  If more than one molecular index is found, their qualities will be concatenated and stored here.")
+    public String MOLECULAR_INDEX_BASE_QUALITY_TAG = "QX";
+
+    @Option(doc="The list of tags to store each molecular index.  If fewer tags are provided than molecular indexes, the remaining molecular indexes " +
```

Comment:
But I will add optional and empty list since I am rusty with the clp parsing conventions in picard.